plumbing/bcf/blockchain_facade.go

```diff
@@ -4,25 +4,37 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/cskr/pubsub"
 	"github.com/filecoin-project/go-filecoin/actor"
-	"github.com/filecoin-project/go-filecoin/actor/builtin"
 	"github.com/filecoin-project/go-filecoin/address"
 	"github.com/filecoin-project/go-filecoin/chain"
 	"github.com/filecoin-project/go-filecoin/exec"
 	"github.com/filecoin-project/go-filecoin/sampling"
 	"github.com/filecoin-project/go-filecoin/state"
 	"github.com/filecoin-project/go-filecoin/types"
 	"github.com/ipfs/go-cid"
-	hamt "github.com/ipfs/go-hamt-ipld"
+	"github.com/ipfs/go-hamt-ipld"
 	"github.com/pkg/errors"
 )
 
+type bcfChainReader interface {
```

Comment:
I like that you're moving away from the ReadStore interface. You should only include the methods that are used though.  It looks like this only needs 'BlockHeight`, `GetBlock`, `GetHead`, `GetTipSetStateRoot`, and `GetTipSet`.